Fix basic introduction notebook

hugobuddel opened the pull request:
The basic introduction notebook was broken, mainly because it did not know about multiple image_planes

hugobuddel commented:
Maybe we can add the jupyter notebooks to the continuous integration?

For MICADO-WISE we plan to use [jupytext](https://github.com/mwouts/jupytext) to couple a normal `.py` file to each `.ipynb` file, and then simply run the Python file. This will not do any checks on the output, but will ensure that the notebook is runnable.

hugobuddel commented:
The notebook is now being tested by adding
```
jupyter nbconvert --execute docs/source/_static/scopesim_basic_intro.ipynb
```
to `.travis.yml`.

This seems to work, e.g. see #28 where it should fail, because there is a 1/0 in the notebook.

Unfortunately, the notebook currently fails on Python 3.5 with:
```
AttributeError: module 'matplotlib.colors' has no attribute 'to_rgba'
```

Perhaps this is because the Python 3.5 version uses an older matplotlib version?

hugobuddel commented:
It seems that `to_rgba()` was introduced in matplotlib 2.1: https://github.com/matplotlib/matplotlib/commit/f6f908e82f5fe231af6de11b1d21e22035a86a58
But matplotlib is fixed on 1.5 in requirements.py35.travis.txt

Maybe it is possible to fix jupyter to the last version supporting matplotlib 1.5.

Or can't we just drop Python 3.5 support? :see_no_evil:

astronomyk commented:
Awesome! Thanks for putting in the notebook tests.
With the py3.5 requirements, I'm aiming to find the "worst" system that
will still run scopesim.
As matplotlib is not really a dependency for scopesim, we can up the
requirement to 2.1 without losing "python parameter space"

Just as an aside, as much as I'd like to drop 3.5 (because then I can use
f"{}" strings :D ), there are enough astronomers who complain about still
wanting to use their py2.7. Hence I think 3.5 needs to be there if we
aren't giving them 2.7. (Py 3.5 was only released in 2015)



-------------------------------------
Kieran Leschinski
Department of Astrophysics
University of Vienna


Am Di., 23. Juni 2020 um 10:40 Uhr schrieb Hugo Buddelmeijer <
notifications@github.com>:

> It seems that to_rgba() was introduced in matplotlib 2.1:
> matplotlib/matplotlib@f6f908e
> <https://github.com/matplotlib/matplotlib/commit/f6f908e82f5fe231af6de11b1d21e22035a86a58>
> But matplotlib is fixed on 1.5 in requirements.py35.travis.txt
>
> Maybe it is possible to fix jupyter to the last version supporting
> matplotlib 1.5.
>
> Or can't we just drop Python 3.5 support? 🙈
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly, view it on GitHub
> <https://github.com/astronomyk/ScopeSim/pull/27#issuecomment-647999036>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACH66QRCYD2T5INHCX6USXLRYBTA3ANCNFSM4OFMXD4Q>
> .
>

hugobuddel commented:
I've now fixed ipykernel to < 4.9.0, which seems to work as well.

This branch should now work, I leave it to you whether you want to merge it as-is (assuming you agree with the changes to the notebook itself) or whether you want to change the matplotlib dependency.

Note that this only checks scopesim_basic_intro.ipynb, which is the one I fixed. I propose to first merge this, and then add checking of the other notebooks in another PR.

Yeah Python 3.5 seems like a good thing to still support. I started using f-strings everywhere in MICADO-WISE, so that is a lost cause...

astronomyk commented:
Happy to add the notbook tests =) 
They all look fine
